content/warnings/uselayouteffect-server-warning.md

```diff
@@ -0,0 +1,47 @@
+---
+title: useLayoutEffect does nothing on the server Warning
+layout: single
+permalink: warnings/uselayouteffect-server-warning.html
+---
+
+You're probably here because you saw this warning in your server logs when rendering a react component.
+
+> Warning:
+>
+> useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client.
+
+`useLayoutEffect` is a hook used to read layout from the dom rerender. They're scheduled to run right after updating the dom, but just before the browser has painted. An example of such a component -
+
+```jsx
+function Foo() {
```

Comment:
I don't think we should use function component names like `Foo` and `Bar`, we should name them to somewhat descriptive component names so there's a better connection with the user reading this document.

Reply:
agreed

Reply:
Great point @trueadm 👍